### PR TITLE
[WFLY-4391] : When testing with WildFly Full, the WSTestCase is randomly unstable

### PR DESCRIPTION
**Pull request by ehsavoie**
It seems that the test relies on WS metrics update which might occurs after the response has been received and processed by the client.
Thus we may get a race.

Jira: https://issues.jboss.org/browse/WFLY-4391

### CODE REVIEW

File: testsuite/integration/smoke/src/test/java/org/jboss/as/test/smoke/webservices/WSTestCase.java [hunk 1]
```diff
@@ -98,16 +107,17 @@ public void testManagementDescription() throws Exception {
         operation.get(ModelDescriptionConstants.RECURSIVE).set(true);
 
         final ModelNode result = managementClient.getControllerClient().execute(operation);
-        Assert.assertEquals(ModelDescriptionConstants.SUCCESS, result.get(ModelDescriptionConstants.OUTCOME).asString());
-
+        assertThat(result.get(ModelDescriptionConstants.OUTCOME).asString(), is(ModelDescriptionConstants.SUCCESS));
```
Comment (kabir) [inline testsuite/integration/smoke/src/test/java/org/jboss/as/test/smoke/webservices/WSTestCase.java]: @ehsavoie all these replacements of Assert.assertEquals() with assertThat() seem pointless, and also make it hard to see exactly what has been changed. If we really 'must' do this replacement, please do so as a separate commit.

Comment (ehsavoie) [inline testsuite/integration/smoke/src/test/java/org/jboss/as/test/smoke/webservices/WSTestCase.java]: Well it helps has you won't get a null assertion for the assertTrue but a more meaningfull message
As for assertEquals I changed those where we are testing for success on operation result
